### PR TITLE
chore(ci): update gateway images in CI

### PR DESCRIPTION
**Pull request by pmalek**
Update images on CI and in setup scripts and include 3.3 in CI matrix.

### CODE REVIEW

File: .github/workflows/integration-test-enterprise.yaml [hunk 1]
```diff
@@ -46,8 +46,9 @@ jobs:
         - '2.7.2.0'
         - '2.8.4.0'
         - '3.0.2.0'
-        - '3.1.1.3'
-        - '3.2.2.1'
+        - '3.1.1.4'
+        - '3.2.2.3'
+        - '3.3.0.0'
```
Comment (aboudreault) [inline .github/workflows/integration-test-enterprise.yaml:51]: is there a reason the series tags (e.g. `kong/kong-gateway:3.1`) ?

**Review by aboudreault — APPROVED** [review]

File: .github/workflows/integration-test-enterprise.yaml (hunk 1 above)
Comment (pmalek) [reply .github/workflows/integration-test-enterprise.yaml]: Good point. I've changed it to use major.minor. PTAL

File: .github/workflows/integration-test.yaml [hunk 2]
```diff
@@ -20,17 +20,18 @@ jobs:
...
+        - '2.3'
+        - '2.4'
+        - '2.5'
+        - '2.6'
+        - '2.7'
+        - '2.8'
+        - '3.0'
+        - '3.1'
+        - '3.2'
+        - '3.3'
```
[the remark this answers is not in the document]
Comment (aboudreault) [reply .github/workflows/integration-test.yaml:34]: I see.  I was trying to pull `kong/kong:2.1` and it was not found for 2.x.

**Review by aboudreault — APPROVED** [review]

File: .github/workflows/integration-test.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (pmalek) [reply .github/workflows/integration-test.yaml:34]: `kong/kong` is something else 😅 https://hub.docker.com/r/kong/kong

> nightly kong images. Official docker hub images can be found here https://hub.docker.com/_/kong

**Pull request merged**